version: "3.8"

services:
  db:
    image: postgres:12.18
    volumes:
      - postgres_data:/var/lib/postgresql/data/
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    ports:
      - 5432:5432
    networks:
      - tracet-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER}"]
      interval: 10s
      retries: 5
      start_period: 30s
      timeout: 5s

  web:
    build:
      context: .
      dockerfile: Dockerfile
    restart: always
    volumes:
      - .:/app
      - /etc/localtime:/etc/localtime:ro
    ports:
      - "8000:8000"
    env_file:
      - .env
    command: bash -c "sleep 10s
      && python manage.py runserver 0.0.0.0:8000"
    depends_on:
      db:
        condition: service_healthy
    networks:
      - tracet-network

volumes:
  postgres_data:

networks:
  tracet-network:
    driver: bridge
